src/importer.ts

```diff
@@ -423,24 +425,21 @@ export default class SystemImporter implements Importer {
 
     try {
       // trigger a scan
-      await this.scanner.scanInto({
-        directory: this.ballotImagesPath,
-        prefix: `batch-${batchId}-`,
-        onFileScanned: async (path) => {
-          await this.fileAdded(path)
-        },
-      })
+      debug('scanning starting for batch: %d', batchId)
+      for await (const sheet of this.scanner.scanSheets(
+        this.ballotImagesPath
+      )) {
+        debug('got a ballot card: %o', sheet)
+        await this.cardAdded(sheet, batchId)
+      }
+      debug('scanning completed successfully')
```

Comment:
this is beautiful.

Reply:
And so much code to make this possible. *sigh*